
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>1.4. omics</title>
    
  <link rel="stylesheet" href="../../_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../_static/sphinx-book-theme.40e2e510f6b7d1648584402491bb10fe.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../_static/js/index.3da636dd464baa7582d2.js">

    <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../../_static/jquery.js"></script>
    <script type="text/javascript" src="../../_static/underscore.js"></script>
    <script type="text/javascript" src="../../_static/doctools.js"></script>
    <script type="text/javascript" src="../../_static/language_data.js"></script>
    <script type="text/javascript" src="../../_static/togglebutton.js"></script>
    <script type="text/javascript" src="../../_static/clipboard.min.js"></script>
    <script type="text/javascript" src="../../_static/copybutton.js"></script>
    <script type="text/javascript">var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script type="text/javascript" src="../../_static/sphinx-book-theme.d31b09fe5c1d09cb49b26a786de4a05d.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/x-mathjax-config">MathJax.Hub.Config({"tex2jax": {"inlineMath": [["\\(", "\\)"]], "displayMath": [["\\[", "\\]"]], "processRefs": false, "processEnvironments": false}})</script>
    <script async="async" type="text/javascript" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script type="text/javascript">
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" type="text/javascript" src="../../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="1.5. complexity" href="ovw_complexity.html" />
    <link rel="prev" title="1.3. disentanglement" href="ovw_disentanglement.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../index.html">
  
  <img src="../../_static/logo.png" class="logo" alt="logo">
  
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../intro.html">
   overview 👋
  </a>
 </li>
</ul>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="research_ovws.html">
   1. research_ovws
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_comp_neuro.html">
     1.1. comp neuro
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_transfer_learning.html">
     1.2. transfer learning
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_disentanglement.html">
     1.3. disentanglement
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     1.4. omics
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_complexity.html">
     1.5. complexity
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_interesting_science.html">
     1.6. interesting science
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_dl_theory.html">
     1.7. dl theory
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_scat.html">
     1.8. scattering transform
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_ml_medicine.html">
     1.9. ml in medicine
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_causal_inference.html">
     1.10. causal inference
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_dl_for_neuro.html">
     1.11. dl for neuro
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_uncertainty.html">
     1.12. uncertainty
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_interp.html">
     1.13. interpretability
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="ovw_generalization.html">
     1.14. generalization
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../cs/cs.html">
   2. cs
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/retrieval.html">
     2.1. info retrieval
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/data_structures.html">
     2.2. data structures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/languages.html">
     2.3. languages
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/software.html">
     2.4. software engineering
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/quantum.html">
     2.5. quantum
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/algo.html">
     2.6. algorithms
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/graphs.html">
     2.7. graphs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/databases.html">
     2.8. overview
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/databases.html#sql">
     2.9. sql
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/os.html">
     2.10. os
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/arch.html">
     2.11. architecture
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/reproducibility.html">
     2.12. reproducibility
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../cs/comp_theory.html">
     2.13. cs theory
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../math/math.html">
   3. math
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../math/differential_equations.html">
     3.1. differential equations
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/proofs.html">
     3.2. proofs
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/analysis.html">
     3.3. real analysis
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/linear_algebra.html">
     3.4. linear algebra
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/signals.html">
     3.5. signals
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/optimization.html">
     3.6. optimization
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/calculus.html">
     3.7. calculus
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/chaos.html">
     3.8. chaos
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../math/math_basics.html">
     3.9. math basics
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../stat/stat.html">
   4. stat
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/graphical_models.html">
     4.1. graphical models
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/data_analysis.html">
     4.2. data analysis
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/testing.html">
     4.3. testing
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/causal_inference.html">
     4.4. causal inference
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/info_theory.html">
     4.5. info theory
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/linear_models.html">
     4.6. linear models
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/time_series.html">
     4.7. time series
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../stat/game_theory.html">
     4.8. game theory
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../ml/ml.html">
   5. ml
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/kernels.html">
     5.1. kernels
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/nlp.html">
     5.2. nlp
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/comp_vision.html">
     5.3. computer vision
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/structure_ml.html">
     5.4. structure learning
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/classification.html">
     5.5. classification
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/unsupervised.html">
     5.6. unsupervised
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/deep_learning.html">
     5.7. deep learning
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/feature_selection.html">
     5.8. feature selection
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/learning_theory.html">
     5.9. learning theory
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ml/evaluation.html">
     5.10. evaluation
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../ai/ai.html">
   6. ai
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/search.html">
     6.1. search
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/decisions_rl.html">
     6.2. decisions, rl
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/fairness_sts.html">
     6.3. fairness, sts
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/cogsci.html">
     6.4. cogsci
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/ai_futures.html">
     6.5. ai futures
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/logic.html">
     6.6. logic
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/philosophy.html">
     6.7. philosophy
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/psychology.html">
     6.8. psychology
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../ai/knowledge_rep.html">
     6.9. representations
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../neuro/neuro.html">
   7. neuro
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/disease.html">
     7.1. disease
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/brain_basics.html">
     7.2. brain basics
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/vissci.html">
     7.3. vision
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/comp_neuro.html">
     7.4. comp neuro
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/sensory_input.html">
     7.5. sensory input
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/memory.html">
     7.6. memory
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/motor.html">
     7.7. motor system
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../neuro/development.html">
     7.8. development
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../_sources/notes/research_ovws/ovw_omics.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/csinva/csinva.github.io"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        
        
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#overview">
   1.4.1. overview
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#vocabulary">
     1.4.1.1. vocabulary
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#data">
     1.4.1.2. data
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#code">
     1.4.1.3. code
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#rna-structure-prediction">
   1.4.2. rna structure prediction
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#rna-basics">
     1.4.2.1. rna basics
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#algorithms">
     1.4.2.2. algorithms
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#d-rna-structure-prediction">
     1.4.2.3. 3d rna structure prediction
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#protein-structure-prediction-dna">
   1.4.3. protein structure prediction (dna)
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#protein-basics">
     1.4.3.1. protein basics
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#algorithm-basics">
     1.4.3.2. algorithm basics
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#deep-learning-papers">
     1.4.3.3. deep-learning papers
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#covid">
     1.4.3.4. covid
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#tips-using-viruses-for-treatment">
   1.4.4. TIPs - using viruses for treatment
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#bin-random-papers">
   1.4.5. bin random papers
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="omics">
<h1>1.4. omics<a class="headerlink" href="#omics" title="Permalink to this headline">¶</a></h1>
<p><strong>some papers involving proteins and ml, especially predicting protein structure from dna/rna</strong></p>
<div class="section" id="overview">
<h2>1.4.1. overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<div class="section" id="vocabulary">
<h3>1.4.1.1. vocabulary<a class="headerlink" href="#vocabulary" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><em>oligonucleotide</em> = <em>oligo</em> = short single strands of synthetic DNA or <em>RNA</em></p></li>
</ul>
</div>
<div class="section" id="data">
<h3>1.4.1.2. data<a class="headerlink" href="#data" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="http://ndbserver.rutgers.edu/">nucleic acid database</a> (NDB)</p></li>
<li><p><a class="reference external" href="https://www.rcsb.org/">protein databank</a> (PDB)</p></li>
<li><p>rna</p>
<ul>
<li><p>~~<a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC165459/">RNABase: an annotated database of RNA structures</a>~~ (2003) - no longer maintained</p></li>
<li><p><a class="reference external" href="https://www.pnas.org/content/106/1/97">Accurate SHAPE-directed RNA structure determination | PNAS</a> (deigan et al. 2009) - chemical probing</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="code">
<h3>1.4.1.3. code<a class="headerlink" href="#code" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>rna</p>
<ul>
<li><p><a class="reference external" href="https://github.com/bgruening/galaxy-rna-workbench">Galaxy RNA workbench</a> - many tools including alignment, annotation, interaction</p></li>
<li><p><a class="reference external" href="https://github.com/ViennaRNA/ViennaRNA">viennaRNA</a> - incorporating constraints into predictions</p></li>
<li><p>neural nets</p>
<ul>
<li><p><a class="reference external" href="https://github.com/jaswindersingh2/SPOT-RNA">spot rna</a> / <a class="reference external" href="https://github.com/jaswindersingh2/SPOT-RNA2">spot rna2</a></p>
<ul>
<li><p>uses an Ensemble of Two-dimensional Deep Neural Networks and Transfer Learning</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://github.com/keio-bioinformatics/mxfold2/">mxfold2</a> (sato et al. 2021)</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="rna-structure-prediction">
<h2>1.4.2. rna structure prediction<a class="headerlink" href="#rna-structure-prediction" title="Permalink to this headline">¶</a></h2>
<div class="section" id="rna-basics">
<h3>1.4.2.1. rna basics<a class="headerlink" href="#rna-basics" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>rna does many things</p>
<ul>
<li><p>certain structures of RNA lend themselves to catalytic activities</p></li>
<li><p>tRNA/mRNA convert DNA into proteins</p></li>
<li><p>RNA also serves as the information storage and replication agent in some viruses</p></li>
</ul>
</li>
<li><p>rna components</p>
<ul>
<li><p>4 bases: adenine-uracil (instead of dna’s thymine), cytosine-guanine</p></li>
<li><p>ribose sugar in RNA is more flexible than DNA</p></li>
</ul>
</li>
<li><p>RNA World Hypothesis (Walter Gilbert, 1986) - suggests RNA was precursor to modern life</p>
<ul>
<li><p>later dna stepped in for info storage (more stable) and proteins took over for catalyzing reactions</p></li>
</ul>
</li>
<li><p>rna structure</p>
<ul>
<li><p><img alt="rna_structure" src="../../_images/rna_structure.png" /></p>
<ul>
<li><p>primary - sequence in which the bases are aligned - relatively simple, comes from sequencing</p></li>
<li><p><strong>secondary</strong> - 2d analysis of hydrogen bonds between rna parts (double-strands, hairpins, loops) - most of stabilizing free energy comes from here (unlike proteins, where tertiary is most important)</p>
<ul>
<li><p>most work on “RNA folding” predicts secondary structure from primary structure</p></li>
<li><p>a lot of this comes from hydrogen bonds between pairs (watson-crick edge)</p>
<ul>
<li><p>other parts of the pairs (e.g. the Hoogsteen- / CH-edge and the sugar edge) can also form bonds</p></li>
</ul>
</li>
</ul>
</li>
<li><p>tertiary - complete 3d structure (e.g. bends, twists)</p></li>
</ul>
</li>
<li><p><img alt="Screen Shot 2021-12-01 at 6.33.41 PM" src="notes/research_ovws/../assets/Screen%20Shot%202021-12-01%20at%206.33.41%20PM.png" /></p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://en.wikipedia.org/wiki/RNA-Seq">RNA-Seq - Wikipedia</a> - RNA-Seq uses next-generation sequencing (NGS) to reveal the presence and quantity of RNA in a biological sample at a given moment, analyzing the continuously changing cellular transcriptome</p></li>
</ul>
</div>
<div class="section" id="algorithms">
<h3>1.4.2.2. algorithms<a class="headerlink" href="#algorithms" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://ocw.mit.edu/ans7870/6/6.047/f15/MIT6_047F15_Compiled.pdf">computational bio book ch 10</a> (kellis, 2016)</p>
<ul>
<li><p>2 main approaches to rna folding (i.e. predicting rna structure):</p>
<ul>
<li><p>(1) thermodynamic stability of the molecule</p></li>
<li><p>(2) probabilistic models</p></li>
<li><p>note: can use evolutionary (i.e. phylogenetic) data to improve either of these</p>
<ul>
<li><p>some RNA changes still result in similar structures</p></li>
<li><p>consistent mutations - something mutates but structure doesn’t change (e.g. AU -&gt; G)</p></li>
<li><p>compensatory mutations - two mutations but structure doesn’t change (e.g. AU -&gt; CU -&gt; CG)</p></li>
<li><p>incorporate similarities to other RNAs along with something like zuker algorithm</p></li>
</ul>
</li>
</ul>
</li>
<li><p>thermodynamic stability - uses more domain knowledge</p>
<ul>
<li><p>dynamic programming approach: given energy value for each pair, minimize total energy by pairing up appropriate base pairs</p>
<ul>
<li><p><strong>assumption</strong>: no crossings - given a subsequence <span class="math notranslate nohighlight">\([i,j]\)</span>, there is either no edge connecting to the ith base (meaning it is unpaired) or there is some edge connecting the ith base to the kth base where <span class="math notranslate nohighlight">\(i &lt; k \leq j\)</span> (meaning the ith base is paired to the kth base)</p>
<ul>
<li><p>this induces a tree-structure to theh problem</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.pnas.org/content/77/11/6309.short">nussinov algorithm</a> (1978)</p>
<ul>
<li><p>ignores stacking interactions between neighboring pairs (i.e. assumes there are no pseudo-knots)</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://academic.oup.com/nar/article-abstract/9/1/133/1043226">zuker algorithm</a> (1981) - includes stacking energies</p></li>
</ul>
</li>
</ul>
</li>
<li><p>probabilistic approach - uses more statistical likelihood</p>
<ul>
<li><p>stochastic context-free grammer (SCFG) is like an extension of HMM that incorporates some RNA constraints (e.g. bonds must happen between pairs)</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0168165617315195">Recent advances in RNA folding - ScienceDirect</a> (fallmann et al. 2017)</p>
<ul>
<li><p>algorithmic advances</p>
<ul>
<li><p>algorithmic constraints</p>
<ul>
<li><p>locality - restrict maximal span of base pairs</p></li>
<li><p>add in bonus energies as hard or soft constraints</p></li>
</ul>
</li>
<li><p>rna-rna and rna-protein interactions - there are different algorithms for telling how 2 things will interact</p></li>
<li><p>newer algorithms take into account some aspects of tertiary structure to predict secondary structure (e.g. motifs, nonstandard base pairs, pseudo-knots)</p></li>
</ul>
</li>
<li><p>representation</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">...((((((((...)).</span> <span class="pre">.)).</span> <span class="pre">.((.((...)).</span> <span class="pre">)))))).</span></code> “dot-parenthesis” notation - opening and closing represent bonded pairs</p></li>
</ul>
</li>
<li><p>evaluation</p>
<ul>
<li><p>computing alignments is non-obvious in these representations</p></li>
<li><p>centroids - structures with a minimum distance to all other structures in the ensemble of possible structures</p></li>
<li><p>consensus structures - given a good alignment of a collection of related RNA structures, can compute their consensus structure, (i.e., a set of base pairs at corresponding alignment positions)</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2982177/">Folding and Finding RNA Secondary Structure</a> (matthews et al. 2010)</p></li>
<li><p>new work drops the assumption of no knots (e.g. crossings)</p>
<ul>
<li><p>project to topology (e.g. low/high genus) - e.g. projecting to a torus can remove crossings but still allow us to use dynamic programming</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.nature.com/articles/s41467-021-21194-4">RNA secondary structure prediction using deep learning with thermodynamic integration</a> (sato et al. 2021)</p>
<ul>
<li><p>RNA-folding scores learnt using a DNN are integrated together with Turner’s nearest-neighbor free energy parameters</p>
<ul>
<li><p>DNN predicts scores that are fed into zuker-style dynamic programming</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="d-rna-structure-prediction">
<h3>1.4.2.3. 3d rna structure prediction<a class="headerlink" href="#d-rna-structure-prediction" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>startup <a class="reference external" href="https://atomic.ai/">atomic ai</a> works on this</p></li>
<li><p><a class="reference external" href="https://www.science.org/doi/10.1126/science.abe5650">Geometric deep learning of RNA structure | science</a> (townshend, …, dror 2021)</p>
<ul>
<li><p>scoring model - gives energy function for (primary sequence, structure) pair</p></li>
<li><p>townshend is at atomic ai</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.frontiersin.org/articles/10.3389/fmolb.2021.720937/full">Frontiers | RNA 3D Structure Prediction Using Coarse-Grained Models | Molecular Biosciences</a> (li &amp; chen, 2021)</p></li>
<li><p>older</p>
<ul>
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/18579566/">iFoldRNA: three-dimensional RNA structure prediction and folding - PubMed</a> (sharma et al. 2008)</p></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="protein-structure-prediction-dna">
<h2>1.4.3. protein structure prediction (dna)<a class="headerlink" href="#protein-structure-prediction-dna" title="Permalink to this headline">¶</a></h2>
<div class="section" id="protein-basics">
<h3>1.4.3.1. protein basics<a class="headerlink" href="#protein-basics" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>the standard way to obtain the 3D structure of a protein is X-ray crystallography</p>
<ul>
<li><p>takes ~1 year &amp;  $120k to obtain the structure of a single protein through X-ray crystallography (<a class="reference external" href="https://fortune.com/2020/11/30/deepmind-protein-folding-breakthrough/">source</a>)</p></li>
</ul>
</li>
<li><p>on average, a protein is composed of 300 amino acids (residues)</p>
<ul>
<li><p>21 amino acid types</p></li>
<li><p>the first residue is fixed</p></li>
</ul>
</li>
<li><p>proteins evolved via mutations (e.g. additions, substitutions)</p>
<ul>
<li><p>fitness selects the best one</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="algorithm-basics">
<h3>1.4.3.2. algorithm basics<a class="headerlink" href="#algorithm-basics" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://www.sciencedirect.com/topics/medicine-and-dentistry/multiple-sequence-alignment">multiple-sequence-alignment</a> (MSA) - alignment of 3 or more amino acid (or nucleic acid) sequences, which show conserved regions within a protein family which are of structural and functional importance</p>
<ul>
<li><p>matrix is (n_proteins x n_amino_acids)</p></li>
<li><p>independent model (PSSM) - model likelihood for each column independently</p></li>
<li><p>pairwise sites (potts model) - model likelihood for pairs of positions</p></li>
</ul>
</li>
<li><p>prediction problems</p>
<ul>
<li><p>contact prediction - binary map for physical contacts in the final protein</p>
<ul>
<li><p>common to predict this, evaluation uses binary metrics</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://arxiv.org/abs/1906.08230">Evaluating Protein Transfer Learning with TAPE</a> - given embeddings, 5 tasks to measure downstream performance (rao et al. 2019)</p>
<ul>
<li><p>tasks: structure (SS + contact), evolutionary (homology), engineering (fluorescence + stability)</p></li>
</ul>
</li>
<li><p>future: interactions between molecules (e.g. protein-protein, environment, highly designed proteins)</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="deep-learning-papers">
<h3>1.4.3.3. deep-learning papers<a class="headerlink" href="#deep-learning-papers" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://www.nature.com/articles/s41586-021-04184-w">De novo protein design by deep network hallucination | Nature</a> (anishchenko…baker, 2021)</p>
<ul>
<li><p>deep networks trained to predict native protein structures from their sequences can be inverted to design new proteins</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.nature.com/articles/s41586-021-03819-2">Highly accurate protein structure prediction with AlphaFold | Nature</a> (jumper, …, hassabis, 2021)</p>
<ul>
<li><p><a class="reference external" href="https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-021-03819-2/MediaObjects/41586_2021_3819_MOESM1_ESM.pdf">supp</a></p></li>
<li><p><a class="reference external" href="https://www.blopig.com/blog/2021/07/alphafold-2-is-here-whats-behind-the-structure-prediction-miracle/">best blog post</a> (<a class="reference external" href="https://towardsdatascience.com/unfolding-alphafold-683d576a54a3">other blog post</a>)</p></li>
<li><p>model overview</p>
<ul>
<li><p><img alt="alphafold" src="../../_images/alphafold.png" /></p></li>
</ul>
</li>
<li><p>preprocessing</p>
<ul>
<li><p>MSA</p></li>
<li><p>finding templates - find similar proteins to model “pairs of residues” - which residues are likely to interact with each other</p></li>
</ul>
</li>
<li><p>evoformer</p>
<ul>
<li><p>uses attention on graph network</p></li>
<li><p>iterative</p></li>
</ul>
</li>
<li><p>structure model - converts msa/pair representations into set of (x,y,z) coordinates</p>
<ul>
<li><p>“invariant point attention” - invariance to translations and rotations</p></li>
</ul>
</li>
</ul>
</li>
<li><p><a class="reference external" href="https://proceedings.mlr.press/v139/rao21a.html">MSA Transformer</a> (rao et al. 2021) - predicts contact prediction</p>
<ul>
<li><p>train via masking</p></li>
<li><p>output (contact prediction) is just linear combination of attention heads finetuned on relatively little data</p></li>
<li><p>some architecture tricks</p>
<ul>
<li><p>attention is axial - applied to rows / columns rather than entire input</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="covid">
<h3>1.4.3.4. covid<a class="headerlink" href="#covid" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4147684/">The Coronavirus Nucleocapsid Is a Multifunctional Protein</a> (mcbride et al. 2014)</p>
<ul>
<li><p>coronavirus nucleocapsid = N protein</p></li>
</ul>
</li>
</ul>
</div>
</div>
<div class="section" id="tips-using-viruses-for-treatment">
<h2>1.4.4. TIPs - using viruses for treatment<a class="headerlink" href="#tips-using-viruses-for-treatment" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p>videos</p>
<ul>
<li><p><a class="reference external" href="https://www.ted.com/talks/leor_weinberger_can_we_create_vaccines_that_mutate_and_spread?language=en#t-798841">Leor Weinberger: Can we create vaccines that mutate and spread? | TED Talk</a></p></li>
<li><p><a class="reference external" href="https://www.youtube.com/watch?v=Dh0RbiAf2jY">Engineering Viruses - Leor Weinberger - Gladstone Institutes 2016 Fall Symposium - YouTube</a></p></li>
</ul>
</li>
<li><p>2 issues</p>
<ul>
<li><p>mutation - viruses mutate, our drugs don’t</p></li>
<li><p>transmission - adherence / deployment - hard to give certain the drugs</p></li>
</ul>
</li>
<li><p>solution: use modified versions of viruses as treatment</p>
<ul>
<li><p><em>therapeutic interfering particles</em> or <em>TIPs</em> are engineered deletion mutants designed to piggyback on a virus and deprive the virus of replication material</p>
<ul>
<li><p>TIP is like a parasite of the virus - it clips off some of the virus DNA (via genetic engineering)</p>
<ul>
<li><p>it doesn’t contain the code for replication, just for getting into the cell</p></li>
<li><p>since it’s shorter, it’s made more efficiently - thus it outcompetes the virus</p></li>
</ul>
</li>
</ul>
</li>
<li><p>how do TIPs spread?</p>
<ul>
<li><p>mutations happen because the “copy machine” within a cell makes a particular mutation</p>
<ul>
<li><p>when a new virus comes along, it makes some of the mutated parts</p></li>
<li><p>TIPs can’t replicate, so they take some of the mutated parts made by the new virus</p></li>
<li><p>then, the TIP gets copied with the same mutation as the virus and this now spreads</p></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
<li><p>effect</p>
<ul>
<li><p>viral load will immediately be lower</p></li>
<li><p>superspreaders can spread treatment to others</p></li>
</ul>
</li>
<li><p><a class="reference external" href="https://www.sciencedirect.com/science/article/pii/S0092867421013192?dgcid=coauthor">Identification of a therapeutic interfering particle—A single-dose SARS-CoV-2 antiviral intervention with a high barrier to resistance</a> (chaturvedi…weinberger, 2021)</p>
<ul>
<li><p>DIP = defective interfering particle = wild-type virus</p></li>
<li><p>single administration of TIP RNA inhibits SARS-CoV-2 sustainably in continuous cultures</p></li>
<li><p>in hamsters, both prophylactic and therapeutic intranasal administration of lipid-nanoparticle TIPs durably suppressed SARS-CoV-2 by 100-fold in the lungs, reduced pro-inflammatory cytokine expression, and prevented severe pulmonary edema</p></li>
<li><p>TIP consists 1k - 2k bases</p></li>
<li><p>hard to actually look at structure here (requires cryoEM)</p></li>
</ul>
</li>
</ul>
</div>
<div class="section" id="bin-random-papers">
<h2>1.4.5. bin random papers<a class="headerlink" href="#bin-random-papers" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li><p><a class="reference external" href="https://www.nature.com/articles/d41586-021-03697-8?utm_source=Nature+Briefing&amp;utm_campaign=b0bda6ba48-briefing-dy-20211210&amp;utm_medium=email&amp;utm_term=0_c9dfd39373-b0bda6ba48-45837550">DeepMind AI tackles one of chemistry’s most valuable techniques</a></p></li>
<li><p><a class="reference external" href="https://www.nature.com/articles/d41586-021-03593-1?utm_source=Nature+Briefing&amp;utm_campaign=b0bda6ba48-briefing-dy-20211210&amp;utm_medium=email&amp;utm_term=0_c9dfd39373-b0bda6ba48-45837550">DeepMind’s AI helps untangle the mathematics of knots</a></p></li>
</ul>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./notes/research_ovws"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="ovw_disentanglement.html" title="previous page">1.3. disentanglement</a>
    <a class='right-next' id="next-link" href="ovw_complexity.html" title="next page">1.5. complexity</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By Chandan Singh<br/>
        
            &copy; Copyright None.<br/>
          <div class="extra_footer">
            <p>
Many of these images are taken from resources on the web.
</p>
          </div>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../_static/js/index.3da636dd464baa7582d2.js"></script>


    
  </body>
</html>